service/history/task/cross_cluster_target_task_executor.go

```diff
@@ -239,37 +239,69 @@ func (t *crossClusterTargetTaskExecutor) executeApplyParentClosePolicyTask(
 		return nil, errMissingTaskRequestAttributes
 	}
 
+	response := types.CrossClusterApplyParentClosePolicyResponseAttributes{}
+	var anyErr error
+
 	for _, childAttrs := range attributes.ApplyParentClosePolicyAttributes {
+		// DON'T RETURN ERROR INSIDE THIS LOOP, CONVERT THEM AND ASSIGN IT TO EACH CHILD'S FailedCause
+		scope := t.metricsClient.Scope(metrics.CrossClusterSourceTaskApplyParentClosePolicyScope)
+		var failedCause *types.CrossClusterTaskFailedCause
+		retriable := false
+
 		targetDomainName, err := t.verifyDomainActive(childAttrs.ChildDomainID)
-		if err != nil {
-			return nil, err
+		if err == nil {
+			err = applyParentClosePolicy(
+				ctx,
+				t.historyClient,
+				childAttrs.ChildDomainID,
+				targetDomainName,
+				childAttrs.ChildWorkflowID,
+				childAttrs.ChildRunID,
+				*childAttrs.ParentClosePolicy,
+			)
 		}
 
-		scope := t.metricsClient.Scope(metrics.CrossClusterSourceTaskApplyParentClosePolicyScope)
-		err = applyParentClosePolicy(
-			ctx,
-			t.historyClient,
-			childAttrs.ChildDomainID,
-			targetDomainName,
-			childAttrs.ChildWorkflowID,
-			childAttrs.ChildRunID,
-			*childAttrs.ParentClosePolicy,
-		)
 		switch err.(type) {
-		case *types.EntityNotExistsError, *types.WorkflowExecutionAlreadyCompletedError, *types.CancellationAlreadyRequestedError:
+		case *types.EntityNotExistsError,
+			*types.WorkflowExecutionAlreadyCompletedError,
+			*types.CancellationAlreadyRequestedError:
 			// expected error, no-op
-			break
+			scope.IncCounter(metrics.ParentClosePolicyProcessorSuccess)
+		case nil:
+			scope.IncCounter(metrics.ParentClosePolicyProcessorSuccess)
 		default:
 			scope.IncCounter(metrics.ParentClosePolicyProcessorFailures)
-			return nil, err
+			failedCause, retriable = t.convertErrorToFailureCause(err)
+			// return error only if the error is retriable
+			if retriable {
+				anyErr = err
+			}
+		}
+		// Optimize SOURCE SIDE RETRIES by returning each child status separately
+		if err != nil {
+			response.FailedChildren = append(
+				response.FailedChildren,
+				&types.ApplyParentClosePolicyResult{
+					Child:       childAttrs,
+					FailedCause: failedCause,
```

Comment:
talked offline, it might be worth to have another iteration to add status to the request and update the request for the target side retries. 

this way we avoid false positive success.